packages/plugins/documentation/__tests__/build-component-schema.test.js

```diff
@@ -0,0 +1,321 @@
+'use strict';
+
+const buildComponentSchema = require('../server/services/helpers/build-component-schema');
+const strapi = require('../__mocks__/strapi');
+
+describe('Build Component Schema', () => {
+  it('builds the Response schema', () => {
+    global.strapi = strapi;
+
+    const apiMocks = [
+      {
+        name: 'users-permissions',
+        getter: 'plugin',
+        ctNames: ['role'],
+      },
+      { name: 'restaurant', getter: 'api', ctNames: ['restaurant'] },
+    ];
+
+    let schemas;
+    for (const mock of apiMocks) {
+      schemas = {
+        ...schemas,
+        ...buildComponentSchema(mock),
+      };
+    }
+
+    const schemaLength = Object.keys(schemas).length;
+    const [pluginResponseName, apiResponseName] = Object.keys(schemas);
+    const [pluginResponseValue, apiResponseValue] = Object.values(schemas);
+
+    const expectedShape = {
+      properties: {
+        data: {
+          type: 'object',
+          properties: {
+            id: { type: 'string' },
+            attributes: { type: 'object', properties: { test: { type: 'string' } } },
+          },
+        },
+        meta: { type: 'object' },
+      },
+    };
+
+    expect(schemaLength).toBe(2);
+    expect(pluginResponseName).toBe('UsersPermissionsRoleResponse');
+    expect(apiResponseName).toBe('RestaurantResponse');
+    expect(pluginResponseValue).toStrictEqual(expectedShape);
+    expect(apiResponseValue).toStrictEqual(expectedShape);
+  });
+
+  it('builds the ResponseList schema', () => {
+    global.strapi = {
+      ...strapi,
+      plugins: {
+        'users-permissions': {
+          ...strapi.plugins['users-permissions'],
+          routes: {
+            'content-api': { routes: [{ method: 'GET', path: '/test', handler: 'test.find' }] },
+          },
+        },
+      },
+      api: {
+        restaurant: {
+          ...strapi.api.restaurant,
+          routes: {
+            restaurant: { routes: [{ method: 'GET', path: '/test', handler: 'test.find' }] },
+          },
+        },
+      },
+    };
+
+    const apiMocks = [
+      {
+        name: 'users-permissions',
+        getter: 'plugin',
+        ctNames: ['role'],
+      },
+      { name: 'restaurant', getter: 'api', ctNames: ['restaurant'] },
+    ];
+
+    let schemas;
```

Comment:
I know this doesn't throw an error or anything, but it would make more sense to me if it was initialized as an empty object